src/de.rs

```diff
@@ -56,16 +136,19 @@ pub fn from_bytes<'a, T>(b: &'a [u8]) -> Result<T>
 where
     T: Deserialize<'a>,
 {
-    let mut deserializer = Deserializer::from_bytes(b);
-    let t = T::deserialize(&mut deserializer)?;
-    if deserializer.input.is_empty() {
-        Ok(t)
-    } else {
-        Err(Error::SyntaxError(deserializer.input[0], None))
-    }
+    _from_slice::<T, Simple>(b)
+}
+/// The same as [`from_bytes`] but `deserialize_any`
+/// will deserialize byte string as `str` if
+/// input bytes are valid UTF-8, otherwise as `bytes`
```

Comment:
... thus making ipossible to deserialize advanced Serde constructs like tagged enums with some penalty for simpler use cases.